src/components/SearchableDropdown/index.tsx

```diff
@@ -12,22 +12,20 @@ const { gray, white, blue } = uiColors;
 
 interface SearchableDropdownProps<T> {
   label: string | React.ReactNode;
-  value: string | T | string[] | T[];
-  onChange: (value: string | T | string[] | T[]) => void;
-  searchFunc?: (
-    value: string | T | string[] | T[],
-    match: string | T
-  ) => boolean;
+  value: T | T[];
+  onChange: (value: T | T[]) => void;
+  searchFunc?: (options: T[], match: string) => T[];
   searchPlaceholder?: string;
   valuePlaceholder?: string;
   options: string[] | Array<T>;
```

Comment:
If we omit string[], we may be able to clean up the if statement block on line 119

Reply:
`string[]` needs to be supplied in order to support the default filtering behavior of the component. Otherwise `searchFunc` would need to be required for this component to work. Since the default `filter` function is not defined on the `T[]` type, typescript will throw an error. So we need to explicitly specify that theres the possibility for `options` to be a `string[]` I added some error handling for that edge case.